File: content/problems/3128-right-triangles/solutions/row-column-one-count-products/Solution.java
class Solution {
    public long numberOfRightTriangles(int[][] grid){
        int rows=grid.length,cols=grid[0].length;
        int[] rowOnes=new int[rows],colOnes=new int[cols];
        for(int row=0;row<rows;row++)for(int col=0;col<cols;col++)if(grid[row][col]==1){
            rowOnes[row]++;
            colOnes[col]++;
        }
        long answer=0;
        for(int row=0;row<rows;row++)for(int col=0;col<cols;col++)if(grid[row][col]==1)answer+=(long)(rowOnes[row]-1)*(colOnes[col]-1);
        return answer;
    }
}

File: content/problems/3128-right-triangles/solutions/row-column-one-count-products/Solution.kt
class Solution {
    fun numberOfRightTriangles(grid:Array<IntArray>):Long{
        val rows=grid.size;
        val cols=grid[0].size;
        val rowOnes=IntArray(rows);
        val colOnes=IntArray(cols);
        for(row in 0 until rows)for(col in 0 until cols)if(grid[row][col]==1){
            rowOnes[row]++;
            colOnes[col]++
        };
        var answer=0L;
        for(row in 0 until rows)for(col in 0 until cols)if(grid[row][col]==1)answer+=(rowOnes[row]-1).toLong()*(colOnes[col]-1);
        return answer
    }
}
